node/bin/src/prover_input_generator/mod.rs

```diff
@@ -158,42 +159,44 @@ fn compute_prover_input(
 
     let prover_input_generation_latency =
         PROVER_INPUT_GENERATOR_METRICS.prover_input_generation[&"prover_input_generation"].start();
-    let prover_input = match replay_record.block_context.execution_version {
-        1 => {
-            use zk_ee::{common_structs::ProofData, system::metadata::BlockMetadataFromOracle};
-            use zk_os_forward_system::run::{
-                StorageCommitment, convert::FromInterface, generate_proof_input,
-                test_impl::TxListSource,
-            };
-
-            let initial_storage_commitment = StorageCommitment {
-                root: fixed_bytes_to_bytes32(root_hash).as_u8_array().into(),
-                next_free_slot: leaf_count,
-            };
-
-            let list_source = TxListSource { transactions };
-
-            let bin_path = if enable_logging {
-                zksync_os_multivm::apps::v1::server_app_logging_enabled_path(&app_bin_base_path)
-            } else {
-                zksync_os_multivm::apps::v1::server_app_path(&app_bin_base_path)
-            };
-
-            generate_proof_input(
-                bin_path,
-                BlockMetadataFromOracle::from_interface(replay_record.block_context),
-                ProofData {
-                    state_root_view: initial_storage_commitment,
-                    last_block_timestamp: replay_record.previous_block_timestamp,
-                },
-                tree_view,
-                state_view,
-                list_source,
-            )
-            .expect("proof gen failed")
-        }
-        v => panic!("Unsupported ZKsync OS execution version: {v}"),
-    };
+    let prover_input =
+        match proving_run_execution_version(replay_record.block_context.execution_version) {
+            1 => unreachable!("proving_run_execution_version does not return 1"), // we prove v1 blocks with v2, it's reflected in `proving_run_execution_version`
+            2 => {
```

Comment:
Just to understand better, here we usually would want to support at most 2 versions, right?